app/src/main/java/org/simple/clinic/drugs/selection/dosage/DosageListItem.kt

```diff
@@ -26,21 +27,32 @@ sealed class DosageListItem : ItemAdapter.Item<DosageListItem.Event> {
     override fun layoutResId() = R.layout.prescribed_drug_with_dosage_list_item
 
     override fun render(holder: BindingViewHolder, subject: Subject<Event>) {
+      val context = holder.itemView.context
       val binding = holder.binding as PrescribedDrugWithDosageListItemBinding
 
       binding.dosageTextView.text = protocolDrug.dosage
+      binding.dosageTextView.setTextColor(context.resolveColor(colorRes = R.color.color_on_surface_67))
+
       holder.itemView.setOnClickListener { subject.onNext(Event.DosageClicked(protocolDrug)) }
     }
   }
 
-  object WithoutDosage : DosageListItem() {
+  data class WithoutDosage(val hasExistingPrescription: Boolean) : DosageListItem() {
 
     override fun layoutResId() = R.layout.prescribed_drug_with_dosage_list_item
 
     override fun render(holder: BindingViewHolder, subject: Subject<Event>) {
+      val context = holder.itemView.context
       val binding = holder.binding as PrescribedDrugWithDosageListItemBinding
 
-      binding.dosageTextView.text = holder.itemView.context.getString(R.string.prescribed_drugs_dosage_none)
+      if (hasExistingPrescription) {
+        binding.dosageTextView.setTextColor(context.resolveColor(attrRes = R.attr.colorError))
+        binding.dosageTextView.text = holder.itemView.context.getString(R.string.prescribed_drugs_dosage_remove)
+      } else {
+        binding.dosageTextView.setTextColor(context.resolveColor(colorRes = R.color.color_on_surface_67))
+        binding.dosageTextView.text = holder.itemView.context.getString(R.string.prescribed_drugs_dosage_none)
```

Comment:
We could extract this into a separate method:

```kotlin
fun setDosageTextAndColor(CharSequence, @ColorRes Int) {
  // Set textview properties here
}
```

What do you think?

Reply:
Ah true. Let me make the change.